File: app/src/main/java/com/pluu/support/kakao/KakaoWeekApi.java
package com.pluu.support.kakao;

import android.content.Context;

import com.pluu.support.impl.AbstractWeekApi;
import com.pluu.support.impl.NAV_ITEM;
import com.pluu.webtoon.item.Status;
import com.pluu.webtoon.item.WebToonInfo;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

/**
 * 카카오 페이지 웹툰 Week API
 * Created by PLUUSYSTEM-NEW on 2015-10-31.
 */
public class KakaoWeekApi extends AbstractWeekApi {

	private static final String[] TITLE = new String[]{"월", "화", "수", "목", "금", "토", "일"};
	private final String WEEK_URL = "http://page.kakao.com/main/ajaxCallWeeklyList";
	private int currentPos;

	public KakaoWeekApi(Context context) {
		super(context, TITLE);
	}

	@Override
	public NAV_ITEM getNaviItem() {
		return NAV_ITEM.KAKAOPAGE;
	}

	@Override
	public List<WebToonInfo> parseMain(int position) {
		this.currentPos = position;

		ArrayList<WebToonInfo> list = new ArrayList<>();

		String response;
		try {
			response = requestApi();
		} catch (Exception e) {
			e.printStackTrace();
			return list;
		}

		Document doc = Jsoup.parse(response);
		Elements links = doc.select(".list");

		final Pattern pattern = Pattern.compile("(?<=/home/)\\d+(?=\\?categoryUid)");
		for (Element a : links) {
            String href = a.select("a").attr("href");
            Matcher matcher = pattern.matcher(href);
            if (!matcher.find()) {
                continue;
            }
            WebToonInfo item = new WebToonInfo(matcher.group());
			item.setTitle(a.select(".title").first().text());
			item.setImage(a.select(".thumbnail img").last().attr("src"));

			if (!a.select(".badgeImg").isEmpty()) {
				item.setStatus(Status.UPDATE);
			}
			item.setWriter(a.select(".info ").text().split("•")[1]);
			list.add(item);
		}

		return list;
	}

	@Override
	public String getMethod() {
		return GET;
	}

	@Override
	public String getUrl() {
		return WEEK_URL;
	}

	@Override
	public Map<String, String> getParams() {
		Map<String, String> map = new HashMap<>();
		map.put("navi", "1");
		map.put("day", String.valueOf(currentPos + 1));
		map.put("inkatalk", "0");
		map.put("categoryUid", "10");
		map.put("subCategoryUid", "1000");
		return map;
	}
}

File: app/src/main/java/com/pluu/support/kakao/KakaoWeekApi.kt
package com.pluu.support.kakao

import com.pluu.support.impl.AbstractWeekApi
import com.pluu.support.impl.NAV_ITEM
import com.pluu.support.impl.REQUEST_METHOD
import com.pluu.webtoon.di.NetworkModule
import com.pluu.webtoon.item.Status
import com.pluu.webtoon.item.WebToonInfo
import org.jsoup.Jsoup

/**
 * 카카오 페이지 웹툰 Week API
 * Created by pluu on 2017-04-25.
 */
class KakaoWeekApi(
    networkModule: NetworkModule
) : AbstractWeekApi(networkModule, KakaoWeekApi.TITLE) {
    override val url = "http://page.kakao.com/main/ajaxCallWeeklyList"
    private var currentPos: Int = 0

    override val naviItem: NAV_ITEM = NAV_ITEM.KAKAOPAGE

    override fun parseMain(position: Int): List<WebToonInfo> {
        this.currentPos = position

        val doc = try {
            Jsoup.parse(requestApi())
        } catch (e: Exception) {
            e.printStackTrace()
            return emptyList()
        }

        val pattern = "(?<=/home/)\\d+(?=\\?categoryUid)".toRegex()
        return doc.select(".list")
            .mapNotNull { element ->
                pattern.find(element.select("a").attr("href"))?.let {
                    WebToonInfo(it.value).apply {
                        title = element.select(".title").first().text()
                        image = element.select(".thumbnail img").last().attr("src")

                        if (element.select(".badgeImg").isNotEmpty()) {
                            status = Status.UPDATE
                        }
                        writer = element.select(".info").text().split("•").last().trim()
                    }
                }
            }
    }

    override val method: REQUEST_METHOD = REQUEST_METHOD.GET

    override val params: Map<String, String>
        get() = mapOf(
            "navi" to "1",
            "day" to (currentPos + 1).toString(),
            "inkatalk" to "0",
            "categoryUid" to "10",
            "subCategoryUid" to "1000"
        )

    companion object {
        private val TITLE = arrayOf("월", "화", "수", "목", "금", "토", "일")
    }
}
